src/AutocompleteBadges.js

```diff
@@ -105,16 +189,22 @@ AutocompleteBadges.STATE = {
 	},
 
 	/**
-	 * The list items of text tha will be filtered by input data.
+	 * The list items of text that will be filtered by input data.
 	 * @type {!Array<!Object>}
 	 * @default []
 	 */
 	dataItems: {
 		validator: Array.isArray,
 		valueFn: function() {
 			return [];
-		}
+		},
+		setter: 'setDataItems'
 	}
 };
 
+const keyCodeLeftArrow 		= 37;
```

Comment:
Since these are local constants, can you name them with uppercase letters (like `KEY_CODE_LEFT_ARROW`), so it's easier to detect what they are when reading the code? Otherwise it initially feels like these weren't declared.